### PR TITLE
Add Github Actions test-cice.yml script

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Add Github Actions scripts
- [X] Developer(s): 
    apcraig, phil-blain
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    No testing done except in github actions, https://github.com/apcraig/CICE/runs/1738212948
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:

Similar to recent Icepack changes, alternative to travis-ci.

See #550

### CODE REVIEW

**Review by eclare108213 — APPROVED** [review]
It looks like travis-CI will still be an option.   Is it now off by default?   I'm not sure how this all works, but happy to go with these changes.   Thanks for figuring it out.

File: README.md
Comment (phil-blain) [inline README.md:1]:
```suggestion
[![Travis-CI](https://travis-ci.org/CICE-Consortium/CICE.svg?branch=master)](https://travis-ci.org/CICE-Consortium/CICE)
```

**Review by phil-blain — APPROVED** [review]

**Comment by apcraig** [comment]
@eclare108213, we are still waiting on the situation with Travis-CI.  We can't use it right now because we don't have an allocation.  In the past, all open source project got unlimited resources.  That's no longer the case.  I have requested that we be granted an allocation on Travis-CI but have not heard anything back.   For now, I'm just leaving it "on", but it's not running.  In the future, we may or may not have Travis-CI, but we will have Github Actions as an alternative and we could use either or both if Travis-CI is available again.

**Comment by phil-blain** [comment]
An added benefit is that the GitHub Actions workflow use the conda environment, so we continuously test the conda setup. If for some reason the created conda environment has problems (like the recent OpenMP bug in the GFortran compiler from conda-forge), then we will immediately notice.

**Comment by apcraig** [comment]
@phil-blain, that's also true and a very handy feature and in part why I'd like to get the linux (ubuntu) working as well.

I wonder why GH Actions is not being triggered on this PR.  For the Icepack PR, it didn't initially either, but then it did at some point.  Some things are quite the mystery.  Hopefully, GH Actions will trigger automatically once we get this PR merged.  I would have more confidence if it would be happening now.  I'll merge, and we'll go from there.

**Comment by apcraig** [comment]
GH Actions seems to be triggered as we hoped.  See #553

**Pull request merged**
